internal/librarian/commit_version_analyzer.go

```diff
@@ -40,7 +40,11 @@ func getConventionalCommitsSinceLastRelease(repo gitrepo.Repository, library *co
 		return shouldIncludeForRelease(files, library.SourceRoots, library.ReleaseExcludePaths)
 	}
 
-	return convertToConventionalCommits(repo, library, commits, shouldIncludeFiles)
+	conventionalCommits, err := convertToConventionalCommits(repo, library, commits, shouldIncludeFiles)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert commits to conventional commits for library %q: %w", library.ID, err)
```

Comment:
Why not writing the error log inside `convertToConventionalCommits`? You can avoid an additional layer of unwrapping errors.

Reply:
`generator: failed to get commits for library generator: did not find commit c64fa6012bad2381cab8a3c764c4eb4b8b14258d when iterating`

This part of the error log seems not easy to understand.

It's OK we can refactor later.